routing/errors.go

```diff
@@ -1,23 +1,77 @@
 package routing
 
-import "errors"
+import "github.com/go-errors/errors"
 
-var (
+const (
 	// ErrNoPathFound is returned when a path to the target destination
 	// does not exist in the graph.
-	ErrNoPathFound = errors.New("unable to find a path to " +
-		"destination")
+	ErrNoPathFound errorCode = iota
 
 	// ErrInsufficientCapacity is returned when a path if found, yet the
 	// capacity of one of the channels in the path is insufficient to carry
 	// the payment.
-	ErrInsufficientCapacity = errors.New("channel graph has " +
-		"insufficient capacity for the payment")
+	ErrInsufficientCapacity
 
 	// ErrMaxHopsExceeded is returned when a candidate path is found, but
 	// the length of that path exceeds HopLimit.
-	ErrMaxHopsExceeded = errors.New("potential path has too many hops")
+	ErrMaxHopsExceeded
 
-	// ErrTargetNotInNetwork is returned when a
-	ErrTargetNotInNetwork = errors.New("target not found")
+	// ErrTargetNotInNetwork is returned when a target node not exists in
+	// the current graph.
+	ErrTargetNotInNetwork
+
+	// ErrOutdatedEdge is returned when the routing update already have
+	// been applied.
+	ErrOutdatedEdge
 )
+
+// errorCode represent the error code and is used for compile time check.
+type errorCode uint8
+
+// routerError is a structure that represent the error inside the routing package,
+// this structure carries additional information about error code in order to
+// be able distinguish errors outside of the current package.
+type routerError struct {
+	err  *errors.Error
+	code errorCode
+}
+
+// Error represents errors as the string
+// NOTE: Part of the error interface.
+func (e *routerError) Error() string {
+	return e.err.Error()
+}
+
+// A compile time check to ensure routerError implements the error interface.
+var _ error = (*routerError)(nil)
+
+// err makes an routerError from the given value. If that value is already an
+// error then it will be used directly, if not, it will be passed to
```

Comment:
`go-errrors/errors` package allows as to be able to see the stack of the error if we will decide to use this errors for unit testing some day.